src/GovUk.Education.ExploreEducationStatistics.Data.Api/Services/FrontendService.cs

```diff
@@ -0,0 +1,54 @@
+#nullable enable
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+using GovUk.Education.ExploreEducationStatistics.Common.Model;
+using GovUk.Education.ExploreEducationStatistics.Common.Model.Data;
+using GovUk.Education.ExploreEducationStatistics.Common.Utils;
+using GovUk.Education.ExploreEducationStatistics.Data.Api.Services.Interfaces;
+using GovUk.Education.ExploreEducationStatistics.Data.Services.ViewModels;
+using Microsoft.AspNetCore.Mvc;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Serialization;
+
+namespace GovUk.Education.ExploreEducationStatistics.Data.Api.Services;
+
+public class FrontendService : IFrontendService
+{
+    private readonly IHttpClientFactory _httpClientFactory;
+
+    public FrontendService(IHttpClientFactory httpClientFactory)
+    {
+        _httpClientFactory = httpClientFactory;
+    }
+
+    public async Task<Either<ActionResult, dynamic>> CreateUniversalTable(TableBuilderResultViewModel tableResult,
+        TableBuilderConfiguration configuration,
+        CancellationToken cancellationToken)
+    {
+        var httpClient = _httpClientFactory.CreateClient("PublicApp");
+
+        var request = new JsonNetContent(
+            content: new UniversalTableFormatCreateRequest(tableResult, configuration),
+            settings: new JsonSerializerSettings
+            {
+                ContractResolver = new CamelCasePropertyNamesContractResolver(),
+                NullValueHandling = NullValueHandling.Ignore
+            });
+
+        var response = await httpClient.PostAsync("api/permalink",
```

Comment:
Do we have this endpoint locked down so only the backend can call it? Not sure if there is any risk if this endpoint was open to anyone.

Reply:
No we don't but it can only be called with table data and respond with a transformed table.

I think we've got another ticket looking at protecting all the private API's.